openapi-cli/src/main/java/io/ballerina/openapi/generators/client/BallerinaAuthConfigGenerator.java

```diff
@@ -321,7 +361,15 @@ public ObjectFieldNode getApiKeyMapClassVariable() { // return ObjectFieldNode
             parameters.add(createToken(COMMA_TOKEN));
             parameters.add(serviceURLNode);
         } else {
-            if (apiKey) {
+            if (apiKey && httpOROAuth) {
```

Comment:
Shall we add an example code snippet in this function documentation for `apiKey && httpOROAuth` scenario ?

Reply:
Done